test/libsolidity/semanticTests/builtins/balance_other_contract.sol

```diff
@@ -0,0 +1,23 @@
+contract Other {
+    constructor() payable {
+    }
+    function getAddress() public returns (address) {
+        return address(this);
+    }
+}
+contract ClientReceipt {
+    Other other;
+    constructor() payable {
+        other = new Other{value:500}();
+    }
+    function getAddress() public returns (address) {
+        return other.getAddress();
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// constructor(), 2000 wei ->
+// contract.balance -> 1500
+// getAddress() -> 0xF01F7809444BD9A93A854361C6FAE3F23D9E23DB
+// contract.balance: 0xF01F7809444BD9A93A854361C6FAE3F23D9E23DB -> 500
```

Comment:
Why is this uppercase btw?

Reply:
I think I just put it in uppercase here. `getAddress()` returned here a decimal encoded address initially, so I converted this address manually to hex. However, I also prefer lowercase hex, so good that you pointed that out.

Reply:
Ah okay, that makes sense. We should prefer the lowercase though :)

Reply:
Yep :)